hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OmBucketReadWriteKeyOps.java

```diff
@@ -0,0 +1,309 @@
+  private int numOfWriteOperations;
+
+  @Option(names = {"-o", "--om-service-id"},
+      description = "OM Service ID"
+  )
+  private String omServiceID = null;
+
+  @CommandLine.Mixin
+  private FreonReplicationOptions replication;
+
+  private Timer timer;
+
+  private ContentGenerator contentGenerator;
+
+  private Map<String, String> metadata;
+
+  private ReplicationConfig replicationConfig;
+
+  private OzoneBucket bucket;
+
+  private int readThreadCount;
+  private int writeThreadCount;
+
+  @Override
+  public Void call() throws Exception {
+    init();
+
+    readThreadCount = (readThreadPercentage * totalThreadCount) / 100;
+    writeThreadCount = totalThreadCount - readThreadCount;
+
+    print("volumeName: " + volumeName);
+    print("bucketName: " + bucketName);
+    print("keyCountForRead: " + keyCountForRead);
+    print("keyCountForWrite: " + keyCountForWrite);
+    print("keySizeInBytes: " + keySizeInBytes);
+    print("bufferSize: " + bufferSize);
+    print("totalThreadCount: " + totalThreadCount);
+    print("readThreadPercentage: " + readThreadPercentage);
+    print("writeThreadPercentage: " + (100 - readThreadPercentage));
+    print("readThreadCount: " + readThreadCount);
+    print("writeThreadCount: " + writeThreadCount);
+    print("numOfReadOperations: " + numOfReadOperations);
+    print("numOfWriteOperations: " + numOfWriteOperations);
+    print("omServiceID: " + omServiceID);
+
+    OzoneConfiguration ozoneConfiguration = createOzoneConfiguration();
+    replicationConfig = replication.fromParamsOrConfig(ozoneConfiguration);
+
+    contentGenerator = new ContentGenerator(keySizeInBytes, bufferSize);
+    metadata = new HashMap<>();
+
+    try (OzoneClient rpcClient = createOzoneClient(omServiceID,
+        ozoneConfiguration)) {
+      ensureVolumeAndBucketExist(rpcClient, volumeName, bucketName);
+      bucket = rpcClient.getObjectStore().getVolume(volumeName)
+          .getBucket(bucketName);
+
+      timer = getMetrics().timer("key-create");
+
+      runTests(this::mainMethod);
+    }
+
+    return null;
+  }
+
+  private void mainMethod(long counter) throws Exception {
+
+    int readResult = readOperations();
+    int writeResult = writeOperations();
+
+    print("Total Keys Read: " + readResult);
+    print("Total Keys Written: " + writeResult * keyCountForWrite);
+
+    // TODO: print read/write lock metrics (HDDS-6435, HDDS-6436).
+  }
+
+  private int readOperations() throws Exception {
+
+    // Create keyCountForRead (defaultValue = 100) keys under
+    // rootPath/readPath
+    String readPath = "".concat(OzoneConsts.OM_KEY_PREFIX).concat("readPath");
+    createKeys(readPath, keyCountForRead);
+
+    // Start readThreadCount (defaultValue = 90) concurrent read threads
+    // performing numOfReadOperations (defaultValue = 50) iterations
+    // of read operations (bucket.listKeys(readPath))
+    ExecutorService readService = Executors.newFixedThreadPool(readThreadCount);
```

Comment:
It looks like `OmBucketReadWriteFileOps` and `OmBucketReadWriteKeyOps` has similar behaviors except the FS and ObjectStore specific apis. How about to keep the common code to an abstract class and expose specific functions. Those functions can be overridden by `fileOps` and `keyOps` specific classes.

```
AbstractOmBucketReadWriteOps 
    | -> OmBucketReadWriteFileOps : Here need to initialize fs and override r/w methods
    | -> OmBucketReadWriteKeyOps : Here need to initialize obs and override r/w methods
```
```

abstract class AbstractOmBucketReadWriteOps  extends BaseFreonGenerator
    implements Callable<Void>{

// Here we can keep all the common functions

// Then expose abstract methods like,

// for specific write logic
protected createKey(String path);

// for specific read logic
protected int readOperations();

}
```

Reply:
Thank you for the suggestions! I have made the necessary changes!